p2p/pkg/rpc/validator/service_test.go

```diff
@@ -286,47 +286,58 @@ func TestProcessEpoch(t *testing.T) {
 	slotDuration := 12 * time.Second
 	svc := validatorapi.NewService(ts.URL, mockValidatorRouter, logger, optsGetter, mockNotifier, notifyOffset)
 
-	svc.SetGenesisTime(time.Now().Add(100*time.Millisecond - slotDuration))
```

Comment:
lets say current time=20s. Then before this PR, genesis_time=8.1s. Upcoming epoch time would be 20.1s, meaning notification time would be 19.1s. Due to this, we were hitting the "notification time already passed for slot", and the notification wasn't firing. 

With new value, genesis_t would be =9.5s, and notification time=20.5s. This lands us in the middle of the 1s notification timeout